pypeit/core/wavecal/wvutils.py

```diff
@@ -199,81 +204,81 @@ def get_wave_grid(waves, masks=None, wave_method='linear', iref=0, wave_grid_min
-        wave_grid_min = waves[masks].min()
-    if wave_grid_max is None:
-        wave_grid_max = waves[masks].max()
-
-    dwave_data, dloglam_data, resln_guess, pix_per_sigma = get_sampling(waves)
-
-    # TODO: These tests of the string value should not use 'in', they should use ==
-    if ('velocity' in wave_method) or ('log10' in wave_method):
-        if dv is not None and dloglam is not None:
-            msgs.error('You can only specify dv or dloglam but not both')
-        elif dv is not None:
-            dloglam_pix = dv/c_kms/np.log(10.0)
-        elif dloglam is not None:
-            dloglam_pix = dloglam
-        else:
-            dloglam_pix = dloglam_data
-
-        # Generate wavelength array
-        wave_grid = wavegrid(wave_grid_min, wave_grid_max, dloglam_pix,
-                             spec_samp_fact=spec_samp_fact, log10=True)
-        loglam_grid_mid = np.log10(wave_grid) + dloglam_pix*spec_samp_fact/2.0
-        wave_grid_mid = np.power(10.0, loglam_grid_mid)
-        dsamp = dloglam_pix
+    if 'user_input' in wave_method:
+        wave_grid = wave_grid_input
+    else:
+        if masks is None:
+            masks = waves > 1.0
+
+        if wave_grid_min is None:
+            wave_grid_min = waves[masks].min()
+        if wave_grid_max is None:
+            wave_grid_max = waves[masks].max()
+
+        dwave_data, dloglam_data, resln_guess, pix_per_sigma = get_sampling(waves)
+
+        # TODO: These tests of the string value should not use 'in', they should use ==
+        if ('velocity' in wave_method) or ('log10' in wave_method):
+            if dv is not None and dloglam is not None:
+                msgs.error('You can only specify dv or dloglam but not both')
+            elif dv is not None:
+                dloglam_pix = dv/c_kms/np.log(10.0)
+            elif dloglam is not None:
+                dloglam_pix = dloglam
+            else:
+                dloglam_pix = dloglam_data
+
+            # Generate wavelength array
+            wave_grid, wave_grid_mid, dsamp = wavegrid(wave_grid_min, wave_grid_max, dloglam_pix,
+                                 spec_samp_fact=spec_samp_fact, log10=True)
+
+        elif 'linear' in wave_method: # Constant Angstrom
+            if dwave is not None:
+                dwave_pix = dwave
+            else:
+                dwave_pix = dwave_data
+            # Generate wavelength array
+            wave_grid, wave_grid_mid, dsamp = wavegrid(wave_grid_min, wave_grid_max, dwave_pix, spec_samp_fact=spec_samp_fact)
+
+        elif 'concatenate' in wave_method:  # Concatenate
+            # Setup
+            loglam = np.log10(waves) # This deals with padding (0's) just fine, i.e. they get masked..
+            nexp = waves.shape[1]
+            newloglam = loglam[:, iref]  # Deals with mask
+            # Loop
+            for j in range(nexp):
+                if j == iref:
+                    continue
+                #
+                iloglam = loglam[:, j]
```

Comment:
A few inline docs would be helpful

Reply:
I didn't make any changes here, I just reordered things in the module.